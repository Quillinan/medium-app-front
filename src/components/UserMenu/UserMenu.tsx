import { useAuth } from '@auth/useAuth';
import IconButton from '@components/IconButton/IconButton';
import { useNavigate } from 'react-router-dom';

const UserMenu: React.FC = () => {
  const { removeAuth } = useAuth();
  const navigate = useNavigate();

  const logout = () => {
    removeAuth();
    navigate('/login');
  };

  return (
    <div className='hidden md:block'>
      <div className='ml-4 flex items-center md:ml-6'>
        <IconButton
          onClick={logout}
          icon={
            <img
              className='h-8 w-8 rounded-full'
              src='https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80'
              alt='User'
            />
          }
        />
      </div>
    </div>
  );
};

export default UserMenu;
