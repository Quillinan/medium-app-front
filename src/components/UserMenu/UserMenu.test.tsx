import { fireEvent, render, screen } from '@testing-library/react';
import UserMenu from './UserMenu';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '@auth/useAuth';
import { Mock } from 'vitest';

vi.mock('react-router-dom', () => ({
  useNavigate: vi.fn(),
}));

vi.mock('@auth/useAuth', () => ({
  useAuth: vi.fn(),
}));

describe('UserMenu', () => {
  const mockNavigate = vi.fn();
  const mockRemoveAuth = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();

    (useNavigate as Mock).mockReturnValue(mockNavigate);
    (useAuth as Mock).mockReturnValue({
      removeAuth: mockRemoveAuth,
    });

    sessionStorage.clear();
  });

  it('should render UserMenu with user name if name is stored in sessionStorage', () => {
    sessionStorage.setItem('name', 'João');

    render(<UserMenu />);

    const userNameElement = screen.getByText('João');
    expect(userNameElement).toBeInTheDocument();

    const logoutButton = screen.getByRole('button');
    expect(logoutButton).toBeInTheDocument();
  });

  it('should render UserMenu with default text if no name is stored', () => {
    render(<UserMenu />);

    const defaultTextElement = screen.getByText('Usuário');
    expect(defaultTextElement).toBeInTheDocument();

    const logoutButton = screen.getByRole('button');
    expect(logoutButton).toBeInTheDocument();
  });

  it('should call removeAuth on successful logout', () => {
    sessionStorage.setItem('name', 'João');

    render(<UserMenu />);

    const logoutButton = screen.getByRole('button');

    fireEvent.click(logoutButton);

    expect(mockRemoveAuth).toHaveBeenCalledTimes(1);

    expect(mockNavigate).toHaveBeenCalledWith('/login');
  });

  it('should match snapshot', () => {
    render(<UserMenu />);

    const { asFragment } = render(<UserMenu />);
    expect(asFragment()).toMatchSnapshot();
  });
});
