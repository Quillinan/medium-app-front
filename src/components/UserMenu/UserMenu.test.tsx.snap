// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserMenu > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="hidden md:block"
  >
    <div
      class="ml-4 flex items-center md:ml-6"
    >
      <button
        class="relative rounded-full bg-gray-800 p-1 text-gray-400 hover:text-white focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
        type="button"
      >
        <span
          class="sr-only"
        >
          Button
        </span>
        <img
          alt="User"
          class="h-8 w-8 rounded-full"
          src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
